src/averageAges.js

```diff
@@ -37,7 +44,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people.filter(mother => {
+    const motherHasChild = (people.some(child => {
+      return child.mother === mother.name;
+    }));
+    const isFemale = mother.sex === 'f';
+
+    return (withChildren)
+      ? motherHasChild
+      : isFemale;
+  });
+
+  const milfsAge = mothers.map(milf => milf.died - milf.born);
```

Comment:
Fix Valerii's comment about variables naming, such jokes will not help you to find a job.